STL_Extension/include/CGAL/Spatial_lock_grid_3.h

```diff
@@ -334,7 +332,7 @@ class Spatial_lock_grid_base_3
   Spatial_lock_grid_base_3(const Bbox_3 &bbox,
                                           int num_grid_cells_per_axis)
     : m_num_grid_cells_per_axis(num_grid_cells_per_axis),
-      m_tls_grids(boost::bind(init_TLS_grid, num_grid_cells_per_axis))
+      m_tls_grids([num_grid_cells_per_axis](){ return init_TLS_grid(num_grid_cells_per_axis); })
```

Comment:
@cjamin @lrineau do you know why a function should be passed  instead of simply passing the value?

Reply:
Not really. The type of `m_tls_grids` is [`tbb::enumerable_thread_specific<bool*, Alloc, Key_type>`](https://software.intel.com/content/www/us/en/develop/documentation/tbb-documentation/top/intel-threading-building-blocks-developer-reference/thread-local-storage/enumerablethreadspecific-template-class.html). Maybe the constructor from a value (`bool*`) was not available in older versions of TBB.

Reply:
It guess it's because the value returned should be retrieved in a thread safe env. You don't want 2 threads to have the same bool value.